website/docs/r/org.html.markdown

```diff
@@ -3,16 +3,20 @@ layout: "vcd"
 page_title: "vCloudDirector: vcd_org"
 sidebar_current: "docs-vcd-resource-org"
 description: |-
-  Provides a vCloud Director Organization resource. This can be used to create and delete an organization.
+  Provides a vCloud Director Organization resource. This can be used to create  delete, and update an organization.
 ---
 
 # vcd\_org
 
-Provides a vCloud Director Org resource. This can be used to create and delete an organization.
+Provides a vCloud Director Org resource. This can be used to create, update, and delete an organization.
 Requires system administrator privileges.
 
 Supported in provider *v2.0+*
 
+!> **Warning:** Up to version 2.4, there were two bugs in the handling of this resource. If you have existing resources
+created in versions 2.0 to 2.4, you should re-create them by following *Upgrading Org resources to 2.5* below.
```

Comment:
Better, but still too wordy and not specific enough. Suggestion to simplify and keep the details in that other section:

**Warning:** If you have existing vcd_org resources in state file created with versions earlier than 2.5, you should re-create them by following *Upgrading Org resources to 2.5* section below.

Reply:
It is not that you should. I have successfully tested creating it in 2.4.0 and updating provider with this PR. It did not even suggest to do any updates. Only in rare case it misbehaves should someone update. In my case it went smooth which is what I like in non major version update. The simple `read` fixed previously incorrect ID (used from task).

Reply:
I have changed the text as suggested, and toned down the certainty of failure with a more cautious wording

Reply:
Guys, we need to get to the bottom of this, because as a user after reading "may" I am left feeling uncertain:
- Should I upgrade the state file?
- How to know whether I should?
- What are consequences if I ignore this?

In other words:
- Do we need this warning at all?
- If yes, we need to make it clear **in which cases** the user should follow that state upgrade section.

Let's discuss it here until we reach a plausible formulation.

Reply:
Adding this from Dainius:

> My manual testing of migrations from 2.4.0 to 2.5.0 are smooth now with no need for import. The only case should someone need to update config is if they manually set stored_vm_quota=-1 and deployed_vm_quota=-1 previously and still it goes without triggering updates. If they used default values instead - the upgrade is hassle-free.

Reply:
You know that you need to rebuild when you get a plan that suggests updating without any changes. 
I added the warning before I accepted Dainius suggestion to handle some possible inconsistencies in the code, and fixing existing bugs in the same sweep.
Perhaps we can remove the warning and leave just the section that says "in rare cases".
How rare, we don't know, but when I reproduced the inconsistencies manually, the procedure we suggest works fine.
With the latest code, the inconsistencies seem to have disappeared in all known cases.
So, the warning could be seen as overcautious, and could be removed.

Reply:
I would agree that it can be removed. I don't know how to get into this situation now.